fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryUnionPlanBase.java

```diff
@@ -237,6 +240,20 @@ public boolean isStrictlySorted() {
 
     @Override
     public QueryPlan<FDBQueriedRecord<Message>> strictlySorted() {
-        return withChildren(getChildren().stream().map(p -> (RecordQueryPlan)p.strictlySorted()).collect(Collectors.toList()));
+        return withChildren(getChildren().stream().map(p -> GroupExpressionRef.of((RecordQueryPlan)p.strictlySorted())).collect(Collectors.toList()));
+    }
+
+    protected static boolean isReversed(@Nonnull List<Quantifier.Physical> quantifiers) {
+        return quantifiers
+                .stream()
+                .map(Quantifier.Physical::getRangesOver)
+                .flatMap(reference -> reference.getMembers().stream())
+                .map(expression -> {
+                    Verify.verify(expression instanceof RecordQueryPlan);
+                    return (RecordQueryPlan)expression;
+                })
+                .map(QueryPlan::isReverse)
+                .findAny()
```

Comment:
This should be consistent per query and therefore all pans will have them or none. I do agree this could be checked here though.

Reply:
The contract of `QueryPlan::isReverse` may be too weak for this purpose in the most general case. In the context of a `UnionCursor`, what matters is whether to emit higher or lower values from the key comparison function. For this to be possible, all the input cursors need to be compatibly ordered. For that to be the case with the scans we have today, they all need to have the same ordering key and they either all reverse their order or none do.

It would probably be superior to have the ordering key representation itself include whether high values of that key come first or low values. Then the subsumption check would take that into account naturally.

Reply:
follow up issue https://github.com/FoundationDB/fdb-record-layer/issues/1318